### PR TITLE
Squashed KeyEvent Branch

### PR DESCRIPTION
**Pull request by ejcohe22**
Set up key event listener for codeAreas
** Vim Command mode and Vim command lists reset if mouse is used! **
-> Is this a bug or how it is supposed to work @KitchenTable99 ?

While in command mode key presses are stored in an array for VIM command chaining.
Can exit command mode with i,I,a,A

Next:
- Stylize cursor/ indicate command mode
- Implement VIM methods in model to be called by Tab controller.

### CODE REVIEW

File: src/proj6BittingCerratoCohenEllmer/controllers/CodeTabController.java [hunk 1]
```diff
@@ -30,6 +35,50 @@
     private final HashMap<Tab,String> savedPaths = new HashMap<>();
 
     private final DialogHelper dialogHelper = new DialogHelper();
+
```
Comment (KitchenTable99) [inline src/proj6BittingCerratoCohenEllmer/controllers/CodeTabController.java:38]: I like the code. Very readable

File: src/proj6BittingCerratoCohenEllmer/controllers/CodeTabController.java [hunk 2]
```diff
@@ -30,6 +35,50 @@
...
+    private final ArrayList<String> vimCommands = new ArrayList<String>();
+
+    private final EventHandler<KeyEvent> vimHandler = new EventHandler<KeyEvent>() {
+        /**
+         * Route VIM key press events to implemented methods
+         * 
+         * @param key A KeyEvent object that gives information about a Key Stroke
+         */
+        @Override
+        public void handle(KeyEvent key) {
```
Comment (KitchenTable99) [inline src/proj6BittingCerratoCohenEllmer/controllers/CodeTabController.java:52]: I feel like this makes the most sense in the subclass of the tab class as part of the instantiation process.

File: src/proj6BittingCerratoCohenEllmer/controllers/CodeTabController.java [hunk 3]
```diff
@@ -30,6 +35,50 @@
...
+                // must handle all key events to supress key stroke in codeArea
+                // Only read key released to prevent duplicated commands
+                if(key.getEventType().equals(KeyEvent.KEY_RELEASED)){
+                    //store commands for chaining
+                    vimCommands.add(key.getText());
+                    System.out.println("Vim Commands: " + vimCommands.toString());
+                    // 'i','I','a','A'-> edit mode
+                    if("i".equals(key.getText()) || "I".equals(key.getText()) ||
+                            "a".equals(key.getText()) || "A".equals(key.getText()) ){
+                        inVIMCommandMode = false;
```
Comment (KitchenTable99) [inline src/proj6BittingCerratoCohenEllmer/controllers/CodeTabController.java:68]: If in command mode, I type command+q, the window does not quit. I think this is the same bug as the red x button, but we'll probably have to create a workaround.

**Review by KitchenTable99 — APPROVED** [review]
This seems pretty good to me. There are a couple of things that will need to be fixed or moved, but it seems like it's worth adding some TODO comments and merging

**Comment by KitchenTable99** [comment]
Also there should be some css change

**Review by iellms — APPROVED** [review]
Just got the chance to read over everything and it all looks good, I agree with most all of what Caleb says, so we should all be good :)

**Pull request merged**